Change 34025 by rgs@scipion on 2008/06/08 14:00:59

Fix for bug [perl #54934] Attempt to free unreferenced scalar
fiddling with the symbol table
Keep the refcount of the globs generated by PerlIO::via balanced.

Affected files ...

... //depot/perl/ext/PerlIO/via/via.pm#9 edit
... //depot/perl/ext/PerlIO/via/via.xs#17 edit

Differences ...
diff -up perl-5.10.0/ext/PerlIO/via/via.xs.34025 perl-5.10.0/ext/PerlIO/via/via.xs
--- perl-5.10.0/ext/PerlIO/via/via.xs.34025	2007-12-18 11:47:07.000000000 +0100
+++ perl-5.10.0/ext/PerlIO/via/via.xs	2009-01-19 09:15:46.000000000 +0100
@@ -89,7 +89,7 @@ PerlIOVia_method(pTHX_ PerlIO * f, char 
 	    if (!s->fh) {
 		GV *gv = newGVgen(HvNAME_get(s->stash));
 		GvIOp(gv) = newIO();
-		s->fh = newRV_noinc((SV *) gv);
+		s->fh = newRV((SV *) gv);
 		s->io = GvIOp(gv);
 	    }
 	    IoIFP(s->io) = PerlIONext(f);
